src/servers/ZoneServer2016/managers/groupmanager.ts

```diff
@@ -26,11 +26,80 @@ enum GroupErrors {
 export class GroupManager {
   /** "Limbo" for an invite that awaits acceptance or denial */
   pendingInvites: { [characterId: string]: number } = {};
+  soloGroups: { [groupId: number]: Group } = {};
+  groupSync: { [groupId: number]: number } = {};
 
   sendGroupError(server: ZoneServer2016, client: Client, error: GroupErrors) {
     server.sendChatText(client, `[GroupError] ${error}`);
   }
 
+  async syncGroup(server: ZoneServer2016, groupId: number) {
+    const group = await this.getGroup(server, groupId);
+    if (!group) return;
+
+    const now = Date.now();
+    const lastSyncTime = this.groupSync[groupId];
+
+    if (!lastSyncTime || lastSyncTime + 5000 <= now) {
+      this.groupSync[groupId] = now;
+
+      const sendData = {
+        unknownDword1: group.groupId,
+        unknownData1: {
+          groupId: group.groupId,
+          characterId: group.leader
+        },
+        members: Object.values(group.members)
+          .map((member, index) => {
+            const client = server.getClientByCharId(member);
+            const character = client?.character;
+
+            if (!client || !character) return null;
+
+            return {
+              characterId: member,
+              inviteData: {
+                characterId: member,
+                identity: {
+                  characterName: character?.name,
+                  unknownQword1: member
+                }
+              },
+              unknownByte1: character.isAlive ? 0 : -1,
+              position: character?.state.position,
+              rotation: character?.state.rotation,
+              memberId: index,
+              unknownQword2: member
+            };
+          })
+          .filter((m) => m !== null)
+      };
+
+      this.sendDataToGroup(server, group.groupId, "Group.Unknown12", sendData);
+    }
+  }
+
+  async getGroup(
+    server: ZoneServer2016,
+    groupId: number
+  ): Promise<Group | null> {
+    return server._soloMode
+      ? this.soloGroups[groupId]
```

Comment:
You can join with two clients in soloMode so it's useful for testing. Best way to solve it would be to get rid of soloMode all together and always start the server in multiplayer mode with bindings to localhost only

Reply:
well that won't happen, we won't bother solo players to have a mongodb locally